src/graphics/aurora/fadequad.cpp

```diff
@@ -0,0 +1,122 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  The Quad to handle one color fades.
+ */
+
+#include <cfloat>
+#include "src/graphics/aurora/fadequad.h"
+
+#include "src/events/events.h"
+
+namespace Graphics {
+
+namespace Aurora {
+
+FadeQuad::FadeQuad() : Renderable(kRenderableTypeGUIFront),
+	_fadeType(kFadeIn), _r(1.0f), _g(0.0f), _b(0.0f),
+	_opacity(0.0f), _wait(0.0f), _run(0.0f) {
+}
+
+void FadeQuad::getColor(float &r, float &g, float &b) const {
+	r = _r;
+	g = _g;
+	b = _b;
+}
+
+void FadeQuad::setColor(float r, float g, float b) {
+	_r = r;
+	_g = g;
+	_b = b;
+}
+
+void FadeQuad::getWaitTime(float & wait) const {
+	wait = _wait/1000;
+}
+
+void FadeQuad::setWaitTime(float wait) {
+	_wait = wait*1000;
+}
+
+void FadeQuad::getRunTime(float & run) const {
+	run = _run/1000;
+}
+
+void FadeQuad::setRunTime(float run) {
+	_run = run*1000;
+}
+
+void FadeQuad::fadeIn() {
+	_fadeType = kFadeIn;
+	_start = EventMan.getTimestamp();
+	show();
+}
+
+void FadeQuad::fadeOut() {
+	_fadeType = kFadeOut;
+	_start = EventMan.getTimestamp();
+	show();
+}
+
+void FadeQuad::calculateDistance() {
+}
+
+void FadeQuad::render(Graphics::RenderPass pass) {
```

Comment:
You should probably make use of the RenderPass parameter here. Right now, the quad would be rendered twice each frame: one for the opaque objects render pass and then again for the transparent objects render pass.

Look at how src/engines/nwn/gui/legal.cpp does it.
(And in fact, maybe you can even make the NWN legal billboard use this class, then? It's not exactly the same, but maybe...)

As it is, the code currently throws a warning because the pass parameter is unused. Please wrap unused parameters in the UNUSED() macro.